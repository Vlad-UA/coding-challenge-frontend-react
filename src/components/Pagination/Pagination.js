// Core
import React, { PureComponent } from 'react';

// Instruments
import styles from './Pagination.module.scss';

export default class Pagination extends PureComponent {
  render() {
    const { currentPage, openPage } = this.props;
    const pageLink = (currentPage <= 2) ? 1 : currentPage - 1;

    return (
      <div className={styles.Pagination}>
        <button type="button" onClick={() => openPage(1)} disabled={currentPage === 1}>{'<< First'}</button>
        <button type="button" onClick={() => openPage(currentPage - 1)} disabled={currentPage === 1}>{'< Prev'}</button>
        <button type="button" onClick={() => openPage(pageLink)} disabled={currentPage === 1}>{pageLink}</button>
        <button type="button" onClick={() => openPage(pageLink + 1)} disabled={currentPage !== 1}>{pageLink + 1}</button>
        <button type="button" onClick={() => openPage(pageLink + 2)}>{pageLink + 2}</button>
        <button type="button" onClick={() => openPage(currentPage + 1)}>{'Next >'}</button>
      </div>
    );
  }
}

// TODO add propTypes
