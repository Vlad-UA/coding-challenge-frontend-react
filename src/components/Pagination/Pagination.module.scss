.Pagination {
  display: grid;
  grid-template-columns: 70px 70px 30px 30px 30px 70px;
  justify-content: center;
  grid-column-gap: 10px;
}
