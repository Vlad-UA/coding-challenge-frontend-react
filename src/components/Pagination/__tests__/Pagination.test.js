// Core
import React from 'react';

// Test utilities
import renderer from 'react-test-renderer';

// Component
import Pagination from '../Pagination';

describe('Component Pagination:', () => {
  test('Must match snapshot:', () => {
    const pagination = renderer
      .create(<Pagination
        totalRecords={3}
        currentPage={1}
        openPage={jest.fn}
      />)
      .toJSON();

    expect(pagination).toMatchSnapshot();
  });
});
