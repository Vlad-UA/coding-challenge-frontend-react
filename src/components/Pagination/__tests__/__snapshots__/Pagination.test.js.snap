// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Pagination: Must match snapshot: 1`] = `
<div
  className="Pagination"
>
  <div
    className="columns"
  >
    <button
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      &lt;&lt; First
    </button>
    <button
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      &lt; Prev
    </button>
  </div>
  <div
    className="columns"
  >
    <button
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      1
    </button>
  </div>
  <div
    className="columns"
  >
    <button
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      Next &gt;
    </button>
    <button
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      Last &gt;&gt;
    </button>
  </div>
</div>
`;
